### ISSUE TITLE
Instructions unclear, little help needed.

### ISSUE THREAD

**Issue by CanineCuriosity** [TilCreator/Tapas-Comic-Downloader#27]
This tool is just what I'm looking for, but I'm struggling a bit with the installation. I'm a complete noob when it comes to coding, so please talk to me like you would talk to an idiot XD
 
This part is the one giving me trouble:
"Navigate to repo location (needs to be downloaded)"

Python is installed, and I can find a lot of modules with help("modules") command, but that's about it. 
(I'm using Windows 10 and Python 3.9)

Any help is appreciated!

**Comment by TilCreator** [comment]
On https://github.com/TilCreator/Tapas-Comic-Downloader click the green download button, click the download zip button. Then extract the zip to somewhere. Open cmd (found in the searchbar), enter `cd <where you have extracted the zip>` (without the `<>`), then enter `pip install -r requirenments.txt` and after that you are ready to use `python tapas-dl.py <name>` (without the `<>`) to download comics.

**Comment by CanineCuriosity** [comment]
Got an error in cmd, it doesn't seem to recognize the pip command. Python is installed, so maybe I missed something during the setup?
![Capture](https://user-images.githubusercontent.com/72964081/97775985-93ccf480-1b6d-11eb-99af-d9b63c90c8ee.PNG)

**Comment by CanineCuriosity** [comment]
Ignore the previous comment, I think I got it working. I tried to use the script, but it didn't do anything. Am I missing something here?
![Capture1](https://user-images.githubusercontent.com/72964081/97776288-8284e780-1b6f-11eb-851b-b5799a04dee6.PNG)

**Comment by TilCreator** [comment]
All three commands should give at least some output. Are the files (requirenments.txt, requirenments_novels.txt, tapas-dl.py) in that folder (C:\Python)?

**Comment by CanineCuriosity** [comment]
Yeah, that's where I put them. So, it seems like the command prompt doesn't recognize the python commands (which means I probably messed up somewhere when installing it). Is there a way to install these using python's own command prompt?

**Comment by TilCreator** [comment]
Sadly idk, you would have to look into the pip docs for that.
To fix the cmd not recognising python you probably want to search for "Windows Python in PATH" or similar.

**Comment by bap269** [comment]
Help, i am completely helpless when it comes to coding. How can I fix this??
![pip](https://user-images.githubusercontent.com/54578953/98088036-ec78f600-1ebb-11eb-90b4-d48478f0f481.PNG)

**Comment by TilCreator** [comment]
You need to be in the same directory where the files (requirenments.txt, requirenments_novels.txt, tapas-dl.py) are. See the `cd` command.

**Comment by Arleng001** [comment]
Uhmm, how to do that? I've downloaded the zip and stored it in drive D. I just copied and pasted it's location to cmd

**Comment by TilCreator** [comment]
Extract the zip and go to the directory where you extracted it. You will find enough help on how to use the `cd` command on the internet.

**Comment by Arleng001** [comment]
Oh okay, got it thanks😊

**Comment by TilCreator** [comment]
Can I close this issue now?

**Comment by arlyn995** [comment]
yeah, i managed to solve the directory issue but when I try to put the  next code it goes like this. I checked the downloaded zip and it's all complete
![tapas1](https://user-images.githubusercontent.com/54578953/98183668-a49db180-1f43-11eb-89e3-14b5b3ec82d9.PNG)

**Comment by arlyn995** [comment]
I've gotten another error it says file or directory not found? I double checked the downloaded file and everything is in there...
![tapas1](https://user-images.githubusercontent.com/54578953/98184561-a5cfde00-1f45-11eb-9b16-d41fa17b6543.PNG)

**Comment by bap269** [comment]
I have a problem when it comes to downloading the novels from tapas whether i enter the URl it says 'piquery' not found. Can somebody pls tell me how to fix it?
![tapas1](https://user-images.githubusercontent.com/54578953/98185381-7ae68980-1f47-11eb-983e-891377dce7d9.PNG)

**Comment by TilCreator** [comment]
Have you installed the requirements.txt?

**Comment by Arleng001** [comment]
I am sorry for being a total ignorant. By installing you mean running it into cmd? Yes i did i ran it through  cmd.

**Comment by TilCreator** [comment]
I mean those two commands:
```bash
pip install -r requirenments.txt
pip install -r requirenments_novels.txt
```

**Comment by CanineCuriosity** [comment]
Alright, I got python working on cmd. I think. It says the download was succesful.
![Capture3](https://user-images.githubusercontent.com/72964081/98726970-e1343780-239f-11eb-95bc-916d312f47ed.PNG)

In case someone else runs into the same problem, this tip from a random forum did the trick:
![Capture5](https://user-images.githubusercontent.com/72964081/98727034-fad57f00-239f-11eb-8d2e-832a456443be.PNG)

However, I ran into another problem (probably 'cause I'm using the code wrong or something):
![Capture4](https://user-images.githubusercontent.com/72964081/98727156-2a848700-23a0-11eb-93f2-3a499b95da83.PNG)
When I run the code, it doesn't give any indicator that it actually did something. Can you provide an example of a fully functional command? Thanks in advance.

**Comment by TilCreator** [comment]
```
~/git/Tapas-Comic-Downloader on master
❯ python tapas-dl.py Erma
Erma [Erma] (450 pages):                                                                                                                                                                                                                                                                                                                                                     
Found directory, only updating (use -f/--force to disable)                                                                                                                                                                                                                                                                                                                   
Detected comic                                                                                                                                                                                                                                                                                                                                                               
Downloaded 21 of 21 images
```

**Comment by TilCreator** [comment]
What does `python --version` (enter in the cmd) say?

**Comment by Arleng001** [comment]
Yes.. I entered the command for novels...

**Comment by bap269** [comment]
An this is where ive come so far...[ 
![piquery](https://user-images.githubusercontent.com/54578953/98756503-aa910800-2405-11eb-8e8b-947478d93bb0.PNG)
](url)

**Comment by TilCreator** [comment]
You **need** both, not just `requirenments_novels.txt`

**Comment by CanineCuriosity** [comment]
Okay, so, cmd doesn't recognize "python" as a command, but it does recognize "py". Here's what I got so far:
![Capture6](https://user-images.githubusercontent.com/72964081/98835178-c4a00a00-2448-11eb-97d9-143ad6ce9aed.PNG)

So, it seems like I can run the command just fine with that small modification, but there is another error here.

**Comment by TilCreator** [comment]
I have **no** idea what is happening there and the only suggestion I have is to use Linux instead of Windows ^^

**Comment by bap269** [comment]
it says directory name is invalid??
![xuanlong](https://user-images.githubusercontent.com/54578953/98896653-750f1c00-24e4-11eb-9018-b466af3988e6.PNG)

**Comment by TilCreator** [comment]
Try `python tapas-dl.py -c xuanlong`
The `-c` option was specificaly created for windows, because it restricts file name characters. Also have a look at `python tapas-dl.py --help`, that lists all available options.

**Comment by bap269** [comment]
Still says the same though even with -c. I also did some research on how to fix the error and it still says the same...

C:\Tapas-Comic-Downloader-master\Tapas-Comic-Downloader-master>python tapas-dl.py -c Xuanlong
Loading Xuanlong...                                                                                                     Xuanlong: Steel and Fang [Xuanlong] (155 pages):                                                                        
Checking folder...                                                                                                      Traceback (most recent call last):
  File "C:\Tapas-Comic-Downloader-master\Tapas-Comic-Downloader-master\tapas-dl.py", line 131, in <module>
    os.mkdir(savePath)
NotADirectoryError: [WinError 267] The directory name is invalid: 'Xuanlong: Steel and Fang [Xuanlong]'

**Comment by TilCreator** [comment]
Oh, you're right, error on my part, it's fixed now

**Comment by CanineCuriosity** [comment]
Was there an error in the code?

Also, I don't have access to a system with linux, but I could try to set up a Virtual Machine. Can you give me the exact version of linux and python you're running? Also, does the code have some random interactions with VMs that I should be aware of?

Lets see if we can find a fix for windows users :3

**Comment by TilCreator** [comment]
Yes

It shouldn't be hard to install on Windows / Only be easier on Linux if you know what you are doing. But shure, try it. I'm using ArchLinux (has no Versioning) and Python 3.8.6 (But the version really should make no difference if it's greater than 3.0.0). And there are no complications between that script and vms that I'm aware of.

There are plenty of people already using the script on Windows without problems.

**Comment by TilCreator** [comment]
Can this be closed?

**Comment by RainbowGh0st** [comment]
Hello, im little new to this and i cant understand the "execute the script part". I am sorry if this is really supid question.

**Comment by PulitoDelgado** [comment]
I had the same problem with "Unable to create process using 'C:\bin\env python3". I managed to solve it by following [this](https://stackoverflow.com/questions/67765928/unable-to-create-process-using-usr-bin-env) with some adaptations. OP had to search Python in a different folder and use the .exe there, while I could use my regular installation folder just fine. 
Thus,
`C:\Users\liutongjun\AppData\Local\Programs\Python\Python312\python.exe tapas-dl.py -c https://tapas.io/series/Erma`
finally worked for me. Now all that's left is to get the script working for locked chapters, despite me having an account with those chapters bought....
Edit: just got it working. exporting cookies on firefox requires a third party extension, unfortunately. Thanks for the script, TilCreator!
